csvcubed/csvcubed/cli/error_mapping.py

```diff
@@ -0,0 +1,195 @@
+    NoMeasuresDefinedError:
+        "At least one measure must be defined in a cube.",
+    NoUnitsDefinedError:
+        "At least one unit must be defined in a cube.",
+    MoreThanOneObservationsColumnError:
+        "Found {error[actual_number]} observed values columns. Only 1 is permitted.",
+    MoreThanOneMeasureColumnError:
+        "Found {error[actual_number]} measures columns. Only 1 is permitted.",
+    MoreThanOneUnitsColumnError:
+        "Found {error[actual_number]} units columns. Only 1 is permitted.",
+    BothMeasureTypesDefinedError:
+        "Measures defined in multiple locations. Measures may only be defined in one location.",
+    BothUnitTypesDefinedError:
+        "Units defined in multiple locations. Units may only be defined in one location.",
+    ObservationValuesMissing:
+        "Observed values missing in '{error[csv_column_title]}' on rows: {error[row_numbers]}",
+    MissingColumnDefinitionError:
+        "Column '{error[csv_column_title]}' is present in CSV but no configuration could be found.",
+    DuplicateColumnTitleError:
+        "There are multiple CSV columns with the title: '{error[csv_column_title]}'.",
+    ColumnValidationError:
+        (
+            "An error occurred when validating the column '{error[csv_column_title]}':" + linesep + 
+            "{error[error]}"
+        ),
+    ColumnNotFoundInDataError:
+        "Configuration found for column '{error[csv_column_title]}' but no corresponding column found in CSV.",
+    UnknownPydanticValidationError:
+        "An error was encountered when validating the cube. The error occurred in '{error[path]}' "
+        "and was reported as '{error[original_error]}'",
+    UndefinedMeasureUrisError:
+        "The Measure URI {error[undefined_values]} found in the data was not defined in the cube config.",
+    UndefinedUnitUrisError:
+        "The Unit URI {error[undefined_values]} found in the data was not defined in the cube config.",
+    UndefinedAttributeValueUrisError:
+        "The Attribute URI {error[undefined_values]} in column '{error[component][label]}' defined in the cube "
+        "config was not found in the data.",
+    ReservedUriValueError:
+        "The URI value(s) {error[conflicting_values]} conflict with the reserved value: '{error[reserved_identifier]}'.",
+    ConflictingUriSafeValuesError:
+        "A URI collision has been detected in: '{error[component]}'.",
+    ValidationError:
+        "A validation error occurred when validating the cube: '{error[message]}'."
+}
+
+
+def friendly_error_mapping(error: ValidationError) -> str:
+    """
+    Given an Exception / Error it returns an error message that is more user-friendly
+    """
+    try:
+        en = error.__class__.__name__
+        msg = MSG_TEMPLATE[error.__class__]
+        ed = error.as_dict()
+
+        if isinstance(error, SpecificValidationError):
+            error.get_error_url()
+
+        if isinstance(error, ConflictingUriSafeValuesError):
```

Comment:
I'm not 100% sure I like the special-casing here. 

Perhaps you could change `MSG_TEMPLATE` so that it's a `Dict[Type[ValidationError], Callable[[ValidationError], str]` - i.e. a map of the validation error type to a lambda which takes the error and returns a string (the error message). Then we could do all of the necessary mapping and have it controlled from one location.

If you need let's have a chat about this.